### PR TITLE
Clean up dead code in System.IO*

### PR DESCRIPTION
**Pull request by xhoogland**
Removed dead code in S.IO.C.Brotli, S.IO.FS.DriveInfo and S.IO.Pipes for issue #17905 to clean up the source.

### CODE REVIEW

File: src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.cs [hunk 1]
```diff
@@ -20,11 +20,6 @@ public DriveInfo(string driveName)
             _name = NormalizeDriveName(driveName);
         }
 
-        private DriveInfo(SerializationInfo info, StreamingContext context)
-        {
-            throw new PlatformNotSupportedException();
-        }
```
Comment (stephentoub) [inline src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.cs]: Isn't this used via reflection for deserialization?

cc: @ViktorHofer... am I wrong, or are we missing tests?

Comment (stephentoub) [reply src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.cs]: > We have been deleting these private serialization constructors as part serialization cleanups

I'm not following @jkotas: those PRs didn't delete these ctors, it made them throw PNSE, e.g. https://github.com/dotnet/corefx/commit/a82f54bd51e22ec3eb14eb5d8f572f022becb8e3#diff-0de900803b15ed30c60fd6dd12c43013R31

Comment (stephentoub) [reply src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.cs]: Ok. That makes sense.

**Comment by jkotas** [comment]
@xhoogland Thanks!

**Pull request merged**